### PR TITLE
feature: coro http client add timeout Function

### PR DESCRIPTION
**Pull request by helintongh**

### CODE REVIEW

File: tests/test_cinatra.cpp [hunk 1]
```diff
@@ -454,3 +454,42 @@ TEST_CASE("test coro http redirect request") {
...
+TEST_CASE("test coro http request timeout") {
+  http_server server(std::thread::hardware_concurrency());
+  bool r = server.listen("0.0.0.0", "8090");
+  if (!r) {
+    std::cout << "listen failed."
+              << "\n";
+  }
+  server.set_http_handler<GET, POST>(
+      "/", [&server](request &, response &res) mutable {
+        std::this_thread::sleep_for(std::chrono::milliseconds(3000));
```
Comment (qicosmos) [inline tests/test_cinatra.cpp:467]: 超时时间有点长，设置成1000ms吧

File: tests/test_cinatra.cpp [hunk 2]
```diff
@@ -454,3 +454,42 @@ TEST_CASE("test coro http redirect request") {
...
+  });
+  f.wait();
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  coro_http_client client{};
+  std::string uri = "http://127.0.0.1:8090";
+
+  resp_data result = async_simple::coro::syncAwait(client.async_get(uri));
+  CHECK(result.status == 200);
+
+  client.set_timeout(1);
```
Comment (qicosmos) [inline tests/test_cinatra.cpp:485]: 接口改成500ms吧

Comment (helintongh) [reply tests/test_cinatra.cpp:485]: timeout set will support  sec and millisecond settings.add this function and modify unittest.

**Comment by qicosmos** [comment]
https://github.com/qicosmos/cinatra/pull/278

**Pull request merged**